xbmc/platform/posix/threads/ThreadImplPosix.cpp

```diff
@@ -8,172 +8,27 @@
-    //          running the app default ->
-    // e.g.          0                  0    -     0   // default non-root user.
-    // e.g.         +20                 0    -     -20 // if root with rlimits set.
-    const int bestUserSetPriority = appNice - userBestNiceValue; // nice is inverted from prio.
-    return std::min(maxPriority, bestUserSetPriority); //
-  }
-  else
-    // If we fail getting the limit for nice we just assume we can't raise the priority
-    return 0;
-}
-#endif
+#include <pthread.h>
 
 std::unique_ptr<IThreadImpl> IThreadImpl::CreateThreadImpl(std::thread::native_handle_type handle)
 {
   return std::make_unique<CThreadImplPosix>(handle);
 }
 
-CThreadImplPosix::CThreadImplPosix(std::thread::native_handle_type handle)
-  : IThreadImpl(handle), m_threadID(GetCurrentThreadPid_())
+CThreadImplPosix::CThreadImplPosix(std::thread::native_handle_type handle) : IThreadImpl(handle)
 {
 }
 
 void CThreadImplPosix::SetThreadInfo(const std::string& name)
 {
 #if defined(TARGET_DARWIN)
   pthread_setname_np(name.c_str());
-#elif defined(TARGET_LINUX) && defined(__GLIBC__)
-  // mthread must be set by here.
-  pthread_setname_np(m_handle, name.c_str());
-#endif
-
-#ifdef RLIMIT_NICE
-  // get user max prio
-  int userMaxPrio = GetUserMaxPriority(ThreadPriorityToNativePriority(ThreadPriority::HIGHEST));
-
-  // if the user does not have an entry in limits.conf the following
-  // call will fail
-  if (userMaxPrio > 0)
-  {
-    // start thread with nice level of application
-    int appNice = getpriority(PRIO_PROCESS, getpid());
-    if (setpriority(PRIO_PROCESS, m_threadID, appNice) != 0)
-      CLog::Log(LOGERROR, "{}: error {}", __FUNCTION__, strerror(errno));
-  }
 #endif
 }
 
 bool CThreadImplPosix::SetPriority(const ThreadPriority& priority)
 {
-  bool bReturn = false;
-
-  std::unique_lock<CCriticalSection> lockIt(m_criticalSection);
-
-  if (!m_handle)
-    bReturn = false;
-#ifdef RLIMIT_NICE
-  else
-  {
-    // get user max prio given max prio (will take the min)
-    int userMaxPrio = GetUserMaxPriority(ThreadPriorityToNativePriority(ThreadPriority::HIGHEST));
-
-    // keep priority in bounds
-    int prio = ThreadPriorityToNativePriority(priority);
-    if (prio >= ThreadPriorityToNativePriority(ThreadPriority::HIGHEST))
-      prio = userMaxPrio; // this is already the min of GetMaxPriority and what the user can set.
-    if (prio < ThreadPriorityToNativePriority(ThreadPriority::LOWEST))
-      prio = ThreadPriorityToNativePriority(ThreadPriority::LOWEST);
-
-    // nice level of application
-    const int appNice = getpriority(PRIO_PROCESS, getpid());
-    const int newNice = appNice - prio;
-
-    if (setpriority(PRIO_PROCESS, m_threadID, newNice) == 0)
-      bReturn = true;
-    else
-      CLog::Log(LOGERROR, "{}: error {}", __FUNCTION__, strerror(errno));
-  }
-#endif
-
-  return bReturn;
+  CLog::Log(LOGDEBUG, "[threads] setting priority is not supported on this platform");
 }
```

Comment:
`xbmc/platform/posix/threads/ThreadImplPosix.cpp:34:1: warning: control reaches end of non-void function [-Wreturn-type]`

Reply:
Thanks!